api/options.c

```diff
@@ -270,7 +270,7 @@ nlopt_result
 NLOPT_STDCALL nlopt_set_lower_bounds(nlopt_opt opt, const double *lb)
 {
      nlopt_unset_errmsg(opt);
-     if (opt && (opt->n == 0 || lb)) {
+     if (opt && (opt->n > 0) && lb) {
```

Comment:
that's what I've done, except I have to specify that the condition depends on opt->lb as the static analysis is not smart enough to understand that opt-n>0 and opt->lb being valid are codependent events.